lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -318,29 +319,74 @@ defmodule Kernel.ParallelCompiler do
   end
 
   defp write_module_binaries({:compile, path}, timestamp, result) do
-    for {{:module, module}, {binary, map}} <- result do
+    Enum.each(result, fn {module, binary} ->
       full_path = Path.join(path, Atom.to_string(module) <> ".beam")
       File.write!(full_path, binary)
       if timestamp, do: File.touch!(full_path, timestamp)
-
-      {module, binary, map}
-    end
+    end)
   end
 
-  defp write_module_binaries(_output, _timestamp, result) do
-    for {{:module, module}, {binary, map}} <- result, do: {module, binary, map}
+  defp write_module_binaries(_output, _timestamp, _result) do
+    :ok
   end
 
-  defp check_modules(modules) do
+  defp maybe_check_modules(result, schedulers) do
     if :elixir_config.get(:bootstrap) do
-      []
+      binaries = for {{:module, module}, {binary, _map}} <- result, do: {module, binary}
+      {binaries, []}
     else
-      Enum.flat_map(modules, fn {_module, binary, module_map} ->
-        Module.Checker.verify(module_map, binary)
-      end)
+      check_modules(result, schedulers)
+    end
+  end
+
+  defp check_modules(result, schedulers) do
```

Comment:
This looks great but I don't want to add more complexity to this module. We should make the new process structure part of a `Module.Checker` (or perhaps have a `Module.ParallelChecker` which then calls `Module.Checker`).